src/dswx_sar/pre_processing.py

```diff
@@ -292,125 +341,141 @@ def run(cfg):
-                logger.info(f'>> computing {pol} {temp_pol_list}')
-
-            # If coherence is in the list,
-            # then compute the coherence from VVVV, VHVH, VVVH
-            if pol in ['coherence']:
-                temp_pol_list = ['VV', 'VH', 'VVVH']
-                logger.info(f'>> computing coherence {temp_pol_list}')
-
-            temp_raster_set = []
-            for temp_pol in temp_pol_list:
-                filename = \
-                    f'{scratch_dir}/{mosaic_prefix}_{temp_pol}.tif'
-                temp_raster_set.append(dswx_sar_util.read_geotiff(filename))
-
-            if pol in ['ratio']:
-                ratio = pol_ratio(np.squeeze(temp_raster_set[0, :, :]),
-                                  np.squeeze(temp_raster_set[1, :, :]))
-                intensity.append(ratio)
-                logger.info(f'computing ratio {co_pol}/{cross_pol}')
-
-            if pol in ['coherence']:
-                coherence = pol_coherence(
-                    np.squeeze(temp_raster_set[0, :, :]),
-                    np.squeeze(temp_raster_set[1, :, :]),
-                    np.squeeze(temp_raster_set[2, :, :]))
-                intensity.append(coherence)
-                logger.info('computing polarimetric coherence')
-
-            if pol in ['span']:
-                span = np.squeeze(temp_raster_set[0, :, :] +
-                        2 * np.squeeze(temp_raster_set[1, :, :]))
-                intensity.append(span)
-
-        else:
-            logger.info(f'opening {pol}')
-            filename = \
-                    f'{scratch_dir}/{mosaic_prefix}_{pol}.tif'
-            temp_raster = dswx_sar_util.read_geotiff(
-                    filename)
-            intensity.append(np.abs(temp_raster))
-
-    intensity = np.asarray(intensity)
+    del ancillary_reloc
 
     # apply SAR filtering
-    filter_size = processing_cfg.filter.window_size
-    filter_flag = processing_cfg.filter.enabled
-    intensity_filt = []
-
-    for ii, pol in enumerate(pol_list):
-        if filter_flag:
-            filtered_intensity = filter_SAR.lee_enhanced_filter(
-                            np.squeeze(intensity[ii, :, :]),
-                            win_size=filter_size)
-            filtered_intensity[filtered_intensity == 0] = np.nan
-            intensity_filt.append(filtered_intensity)
-        else:
-            filtered_intensity = intensity[ii, :, :]
-            intensity_filt.append(filtered_intensity)
+    pad_shape = (filter_size, 0)
+    block_params = dswx_sar_util.block_param_generator(
+        lines_per_block=line_per_block,
+        data_shape=(im_meta['length'],
+                    im_meta['width']),
+        pad_shape=pad_shape)
+
+    for block_ind, block_param in enumerate(block_params):
+        output_image_set = []
+        for polind, pol in enumerate(pol_list):
+            logger.info(f'block processing {block_ind} - {pol}')
+
+            if pol in ['ratio', 'coherence', 'span']:
```

Comment:
I'm lost here. I thought valid polarizations would be values found in following dict: https://github.com/opera-adt/DSWX-SAR/blob/88498999a18f03a23c614a33955b12cdd2ffde97/src/dswx_sar/dswx_runconfig.py#L21-L30

Can you point me to where polarizations get assigned to any of the following values: `['ratio', 'coherence', 'span']`? My grep of the branch has not yielded anything.

Reply:
`['ratio', 'coherence', 'span']` are the polarimetric measurement computed from the combination of the different polarizations. For example, 'ratio' is the ratio between co- and cross- pol. These are given from the runconfig, but do not need to be included in `DSWX_S1_POL_DICT`.  `DSWX_S1_POL_DICT` are more like pure polarizations which can be observed in RTC images.

Reply:
You may be right. In particular, the second point could cause an issue. So, I introduced the new member in `runconfig`, which is `polarimetric_option'. 
The default is `None` because, in production, we are not going to use it.